### ISSUE TITLE
Transitive dependencies in 0.37.0

### ISSUE THREAD

**Issue by davidrothera** [CocoaPods/CocoaPods#3488]
Since updating to 0.37.0 I'm having issues where the below Podfile gives an error:

Podfile:
```ruby
# Uncomment this line to define a global platform for your project
# platform :ios, '6.0'

# Uncomment this line to define a global platform for your project
platform :ios, '8.0'

# ignore all warnings for all pods
inhibit_all_warnings!

# swift
use_frameworks!

pod 'Stripe'
pod 'PaymentKit'
pod 'Masonry'
pod 'Parse'
pod 'TSMessages'
pod 'ParseUI'
pod 'SVProgressHUD'
pod 'SnapKit'

target 'Stuff' do

end

target 'StuffTests' do

end
```

Error:
```
[!] The 'Pods' target has transitive dependencies that include static binaries: (/Users/drothera/Development/Stuff/Pods/Parse/libParse.a)
```

Rolling back to 0.36.4 works without issues.

**Comment by kylef** [comment]
This message is because you are trying to install a pod with static binaries (`Parse`) as a dynamic library (via `use_frameworks!`) which is not supported. As this message indicates (could probably be clearer).

See https://github.com/CocoaPods/CocoaPods/issues/3267 for more information.

**Comment by segiddins** [comment]
Hm, maybe in this case the exception is incorrect?

**Comment by davidrothera** [comment]
Can you please re-open this and check into it? I understand the message and checked into it with Parse and it works perfectly if you rollback to 0.36.4

**Comment by neonichu** [comment]
This happens because of #3468, both Parse pods are now linked directly to the user target, as they actually always should have been. Unfortunately, the transitive dependencies check is too strict, it needs to take into account that because both Pods are linked directly, there is no issue.

**Comment by neonichu** [comment]
While the transitive dependency check is indeed too strict (see #3531), I missed that ParseUI actually does contain source code, so it is correctly rejected in 0.37.

What should happen is that we would still create a pod target for Parse when integrating as frameworks, but we currently have no way to distinguish if that would work or not (e.g. the GoogleAnalytics' binary can not be used in that way, see #3346).

**Comment by davidrothera** [comment]
This has been resolved in Parse 1.7.2.2
